src/selectors/getChildren.ts

```diff
@@ -74,3 +74,31 @@ export const getChildrenRanked = (state: State, context: Context): Child[] =>
 /** Returns the first visible child of a context. */
 export const firstVisibleChild = (state: State, context: Context) =>
   getChildrenSorted(state, context)[0]
+
+/**
+ * Returns visible children.
+ *
+ * @param state - Redux state.
+ * @param path - Parent's path.
+ * @param simplePath - Parent's simple path.
+ * @param children - Child or ThoughtContext array.
+ * @param showContexts - Parent's context view status.
+ */
+export const filterChildren = (state: State, path: Path, simplePath: SimplePath, children: (Child | ThoughtContext)[], showContexts?: boolean) => {
+  return children.filter(child => {
+
+    const childSimplePath = getChildPath(state, child, simplePath, showContexts)
+    const childPath = appendChildPath(state, childSimplePath, path)
+
+    const value = showContexts
+      ? head((child as ThoughtContext).context)
+      : (child as Child).value
+
+    return state.showHiddenThoughts ||
+      // exclude meta thoughts when showHiddenThoughts is off
+      // NOTE: child.rank is not used by isChildVisible
+      isChildVisible(state, pathToContext(simplePath), { value, rank: child.rank }) ||
```

Comment:
Both `state.showHiddenThoughts` and `isChildVisible` checks are provided used in this module, e.g. `getVisibleThoughts`. You'll want to reuse code rather than duplicating it. Notice how much higher order functions are leveraged in this module in order to make everything composable without duplication. If you need to split up functions so that they are more modular you can do that.